sdk/formrecognizer/azure-formrecognizer-perf/src/main/java/com/azure/formrecognizer/perf/core/ServiceTest.java

```
@@ -0,0 +1,101 @@
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.http.ProxyOptions;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.polling.SyncPoller;
+import com.azure.perf.test.core.PerfStressOptions;
+import com.azure.perf.test.core.PerfStressTest;
+import reactor.core.publisher.Mono;
+
+import java.net.InetSocketAddress;
+
+/**
+ * Base class for Azure Formrecognizer performance tests.
+ */
+public abstract class ServiceTest<TOptions extends PerfStressOptions> extends PerfStressTest<TOptions> {
+    private static final String CONFIGURATION_ERROR = "Configuration %s must be set in either environment variables "
+        + "or system properties.%n";
+
+    protected final FormRecognizerClient formrecognizerClient;
+    protected final FormRecognizerAsyncClient formrecognizerAsyncClient;
+    protected final FormTrainingClient formTrainingClient;
+    private final FormTrainingAsyncClient formTrainingAsyncClient;
+
+    protected String modelId;
+
+    /**
+     * The base class for Azure Formrecognizer tests.
+     *
+     * @param options the configurable options for performing perf testing on this class.
+     */
+    public ServiceTest(TOptions options) {
+        super(options);
+
+        String formrecognizerEndpoint = Configuration.getGlobalConfiguration().get("AZURE_FORMRECOGNIZER_ENDPOINT");
+        if (CoreUtils.isNullOrEmpty(formrecognizerEndpoint)) {
+            System.out.printf(CONFIGURATION_ERROR, "AZURE_FORMRECOGNIZER_ENDPOINT");
+            System.exit(1);
+        }
+
+        String formrecognizerApiKey = Configuration.getGlobalConfiguration().get("AZURE_FORMRECOGNIZER_API_KEY");
+        if (CoreUtils.isNullOrEmpty(formrecognizerApiKey)) {
+            System.out.printf(CONFIGURATION_ERROR, "AZURE_FORMRECOGNIZER_API_KEY");
```

better to throw an exception.